src/PDPVerifier.sol

```diff
@@ -388,6 +382,8 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         uint256 challengeEpoch = nextChallengeEpoch[setId];
         require(block.number >= challengeEpoch, "premature proof");
         require(proofs.length > 0, "empty proof");
```

Comment:
@ZenGround0 We should also revert `provePossession ` if `challengeRange` is 0.

Reply:
Also `provePossession` should also assert that the current epoch > challengeEpoch

Reply:
This check would be correct but I don't think I like this because this check is covered by requiring that challengeEpoch != NO_CHALLENGE_SCHEDULED

The only way to schedule a challenge is to call nextProvingPeriod which sets challengeRange equal to leafCount.  If the leafCount is at zero nextProvingPeriod will revert challengeEpoch back to NO_CHALLENGE_SCHEDULED.

Can you see any edge case around this?  If there is I think I'll want to do some more thinking about the design we've got.

Reply:
Yes but what's the harm in putting this in place to make it explicit ? I can't see an edge case around this but want to make it more explicit here.

Reply:
Your call, no strong opinions.